### PR TITLE
tibble v1.0.0 compatibility

### PR DESCRIPTION
**Pull request by dpprdan**
This PR ensures compatibility with tidyr >= v1.0.0 and backward compatibility with tidyr v0.8.x. I made the following changes: 

- I dropped the `point_list` and `bbox_list` classes, because `unnest()` does not handle custom subclasses well, see https://github.com/tidyverse/tidyr/issues/761. Both subclasses were only used for `print`ing anyway and making that work seems more of a hassle than that is worth.
- `unnest()` in tidyr 1.0 [requires](https://tidyr.tidyverse.org/reference/nest.html) an explicit choice for name repair, or else it will throw an error if there are duplicate column names. I opted for [`names_repair = "unique"`](https://vctrs.r-lib.org/reference/vec_as_names.html#unique-names).
- In addition all columns coming from OpenCage API are now prefixed with "oc_" in order to avoid automatic renaming as much as possible. Another advantage of this is that it is now possible to select all OpenCage columns easily with e.g. `select(xdf, starts_with("oc_"))`. 
- `unnest()` is [implemented differently](https://github.com/ropensci/opencage/commit/c5507f61d3a436f0c197f59eb6fd4403a333fc66), depending on the tidyr version. 
- Fix: list columns are not silently dropped anymore when using the `_df` geocoding functions.

A few other changes only loosely related to the new tidyr release: 
- ensure lintr v2.0.0 compatibility (do we need to empty the package cache on Travis?) 
- The `opencage_` functions now use `tibble::as_tibble()` instead of the deprecated `dplyr::tbl_df()`
- `oc_check_query()` now checks that all provided arguments are either length one or the length of either placename or latitude.

### CODE REVIEW

File: R/deprecated.R [hunk 1]
```diff
@@ -20,14 +20,12 @@ NULL
 #'
 #' @return A list with
 #' \itemize{
-#' \item results as a data.frame (`dplyr` `tbl_df`) called results with one line
-#'  per results,
+#' \item results as a tibble/data.frame with one line per result,
```
Comment (maelle) [inline R/deprecated.R]:
```suggestion
#' \item results as a tibble with one line per result,
```

tibbles are not data.frames according to the longest [R-pkg-devel thread ever](https://stat.ethz.ch/pipermail/r-package-devel/2017q3/001896.html). Hence my not being sure we could use "tibble/data.frame" without making someone angry.

File: R/oc_forward.R [hunk 2]
```diff
@@ -412,29 +416,35 @@ oc_forward_df <-
             )
         )
 
-      results <- tidyr::unnest(results_nest, op) # nolint
-      # `op` is necessary, so that other list columns are not unnested
-      # but lintr complains about `op` not being defined
+      if (utils::packageVersion("tidyr") > "0.8.99") {
+        results <- tidyr::unnest(results_nest, op, names_repair = "unique") # nolint
+        # specifying `op` is necessary, so that other list columns are not
+        # unnested, but lintr complains about `op` not being defined
```
Comment (maelle) [inline R/oc_forward.R]: I had similar issues with another package yesterday, and can't remember whether one can use the rlang .data adverb in such cases too? But fine to leave the lintr exception, it's easier... if R CMD check doesn't complain either.

**Comment by maelle** [comment]
LGTM, thanks a lot both for the code and the summary! My comments are minor so feel free to merge once you've made a decision.

File: R/deprecated.R (hunk 1 above)
Comment (dpprdan) [reply R/deprecated.R]: I wasn't aware of this thread. It is long indeed and ... interesting. Not sure, whether I agree that "tibbles are not data.frames" based on the evidence presented there, but that does not matter and it does no harm to omit "data.frame" here.

File: R/oc_forward.R (hunk 2 above)
Comment (dpprdan) [reply R/oc_forward.R]: This does work with rlang::.data (now), so I made the necessary changes.

**Pull request merged**